homeassistant/components/reolink/config_flow.py

```diff
@@ -113,7 +113,9 @@ async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo) -> FlowRes
                 raise AbortFlow("already_configured")
 
             # check if the camera is reachable at the new IP
-            host = ReolinkHost(self.hass, existing_entry.data, existing_entry.options)
+            new_config = dict(existing_entry.data)
```

Comment:
It seems like this is missing test coverage if it can be changed without updating tests

Reply:
Well there is a check that goes over this code, but since the test controlls if a `ConnectionError` occurs or not it does not mather if the IP is valid or not.
The IP that is eventually saved to the config is checked, but that is correct, see line 142:
`        self._abort_if_unique_id_configured(updates={CONF_HOST: discovery_info.ip})`
Where the correct `discovery_info.ip` is used

Reply:
I just added some extra assertions (`assert_called_with`) in the tests so that it would now fail if this would be changed in the future.